### PR TITLE
Automatically add levels in default directory to list for build selec…

### PR DESCRIPTION
**Pull request by jpmartin2**
#### What features does this propose?
- Adds levels in the ~/.aeneas-kabsuji directory to the list of levels in the BuildSelectLevelView
- Cleaned up a few other things

### CODE REVIEW

File: src/main/java/aeneas/models/LevelIndex.java [hunk 1]
```diff
@@ -19,13 +19,12 @@
...
 
-  public final String defaultLevelPath;
+  public final Path defaultLevelPath;
 
   LevelIndex() {
     levels = new HashMap<>();
     // TODO: Consider changing this path.
-    String homeDir = System.getenv("HOME");
-    defaultLevelPath = homeDir + "/.aeneas-kabasuji";
+    defaultLevelPath = Paths.get(System.getenv("HOME"), ".aeneas-kabasuji");
```
Comment (PeterMitrano) [inline src/main/java/aeneas/models/LevelIndex.java]: This actually needs to check HOMEPATH on windows. Check `System.getProperty("os.name").contains("Windows")`

File: src/main/java/aeneas/views/BuildSelectLevelView.java [hunk 2]
```diff
@@ -101,6 +108,57 @@ public void initialize(URL location, ResourceBundle resources) {
...
     JFXDepthManager.setDepth(fileList, 1);
 
+    refresh();
   }
 
+  public void refresh() {
+    fileList.getItems().clear();
+    
+    fileList.getItems().add(createNewLevelLabel);
+    Path defaultLevelPath = Paths.get(System.getenv("HOME"), ".aeneas-kabasuji");
```
Comment (PeterMitrano) [inline src/main/java/aeneas/views/BuildSelectLevelView.java]: This code is duplicated. I maybe make a method on Model to do some of this.

File: src/main/java/aeneas/views/BuildSelectLevelView.java [hunk 3]
```diff
@@ -101,6 +108,57 @@ public void initialize(URL location, ResourceBundle resources) {
...
+          ois = new ObjectInputStream(new FileInputStream(file.toFile()));
+        } catch (IOException e) {
+          System.err.println("Error reading file '" + file + "'");
+        }
+
+        try {
+          Level l = (Level) ois.readObject();
+          levelMap.put(file.toString(), l);
+        } catch (NumberFormatException e) {
+          // Do something here to load custom levels
```
Comment (PeterMitrano) [inline src/main/java/aeneas/views/BuildSelectLevelView.java]: So how do we want to handle custom levels? At this point, I say we just make any level greater than 15 custom. We can limit it to playing levels in `$HOME/.aeneas-kabasuji`. If you name them correctly they will show up and can be played. Thoughts?

Comment (jpmartin2) [inline src/main/java/aeneas/views/BuildSelectLevelView.java]: I think that's probably the easiest at this point.

File: src/main/java/aeneas/views/BuildSelectLevelView.java (hunk 2 above)
Comment (jpmartin2) [inline src/main/java/aeneas/views/BuildSelectLevelView.java]: Yeah, if we settle for only having numbered levels (where >15 means custom) then I think we can use the same level index for the build select level as well

File: src/main/java/aeneas/views/BuildSelectLevelView.java (hunk 3 above)
Comment (PeterMitrano) [inline src/main/java/aeneas/views/BuildSelectLevelView.java]: Ok. That can come after this in a separate PR. Just look at my other comments and we'll be good

File: src/main/java/aeneas/models/LevelIndex.java (hunk 1 above)
Comment (jpmartin2) [inline src/main/java/aeneas/models/LevelIndex.java]: Looks like `System.getProperty("user.home");` might work for all platforms. We should test it to be sure though.

Comment (PeterMitrano) [inline src/main/java/aeneas/models/LevelIndex.java]: oh nice. I can test linux/windows

Comment (jpmartin2) [inline src/main/java/aeneas/models/LevelIndex.java]: Can confirm it works on mac

**Comment by jpmartin2** [comment]
Hmmm... The build failed with 
`> Unable to read execution data file /home/travis/build/Cheddarpuffs/Aeneas-KabaSuji/build/jacoco/test.exec`

No idea what that means.

Rerunning fixed it though.

**Comment by PeterMitrano** [comment]
There is a retry button on travis btw.

**Comment by jpmartin2** [comment]
Ah, I didn't see that

**Comment by jkuszmaul** [comment]
If we're going to do this, do we really need to open an explicit save dialog? At the very least, the save dialog should open and automatically fill in the correct filename for the given level number (I think that this can be done entirely inside the `MainView`'s open-save-dialog option).

**Comment by jpmartin2** [comment]
I think I'll have it just default to saving the right filename if you're editing an existing level.

**Comment by jpmartin2** [comment]
I just made it so the default path to save a level is the right name if the level was already numbered. 

I also made 'non-prebuilt' mean any level with number > 15, and made them unlocked by default.

**Pull request merged**